dist/systemd/obs-api-support.target

```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description = Open Build Service API Support Daemons
-Wants = obsapisetup.service obs-clockwork.service obs-delayedjob-queue-consistency_check.service obs-delayedjob-queue-default.service obs-delayedjob-queue-issuetracking.service obs-delayedjob-queue-mailers.service obs-delayedjob-queue-project_log_rotate.service obs-delayedjob-queue-quick@0.service obs-delayedjob-queue-quick@1.service obs-delayedjob-queue-quick@2.service obs-delayedjob-queue-releasetracking.service obs-sphinx.service
```

Comment:
Why?

Reply:
> Why?

As far as I know `obsapisetup.service` is a script that is only needed to be run when the obs-api package gets installed. Isn't it?

Reply:
AFAICS it is also used to setup the database

https://github.com/openSUSE/open-build-service/blob/master/dist/setup-appliance.sh#L259

Reply:
but it should only be enabled in the appliance - and configure itself which services to start before. Something you can easily do with systemd, but hardly with sysvinit. But as said: @bugfinder is already redoing the services

Reply:
@coolo: You are right, but i doubt that @bugfinder is working on obsapisetup.

@Ana06: After a closer look into our OBS-Appliance I have seen that we enable the obsapisetup manually while building the appliance.

https://build.opensuse.org/package/view_file/OBS:Server:Unstable/OBS-Appliance/config.sh?expand=1
SEE Line 43

So your change might not break the appliance. Anyway I would like to solve it with a different approach - migrating obsapisetup to systemd (I hope this will solve the mentioned issue too).

This is my approach

https://github.com/M0ses/open-build-service/commit/1550565c0b8d0b37aa23b9c8d1610d3f185a1c05

It has not been tested for the appliance so far but I`m working on it.

@Ana06 What do you think? Could this solve issue #6188?

Reply:
you still want to avoid it in the default presets. So the spec file changes are still required from this PR

Reply:
> you still want to avoid it in the default presets. 

Do we? Maybe I missed the point here, but this would change the default behavior of our packages. So far (IIRC), it was enough to install our packages and change the value of "OBS_API_SETUP=" in /etc/sysconfig/obs-server to "yes" to get an automatic setup on next reboot. This behavior would (IMHO) be changed and see no documentation about that.

Reply:
Where is the documentation for the current behaviour?

Reply:
LOL - In the code? 
Even if we did no explicit documentation in the past - we should not change the default behavior unless there a good reasons for it.
If there are good reasons, I'm ok with changing the default behavior
But not without documentation.

Reply:
I didn't follow the full discussion here, but isn't the good reason that we currently enable a service in our packages that isn't there?

And if we change this (the OBS setup automatism you mentioned), all we have to do is to mention the change in the release notes and point to our admin guide that documents how the OBS_API_SETUP thingie works.

Reply:
> I didn't follow the full discussion here, but isn't the good reason that we currently enable a service in our packages that isn't there?

As already stated in https://github.com/openSUSE/open-build-service/pull/6189#discussion_r231169900 I´m absolutly fine with removing it from the scripts in the spec.

> 
> And if we change this (the OBS setup automatism you mentioned), all we have to do is to mention the change in the release notes and point to our admin guide that documents how the OBS_API_SETUP thingie works.

I´m fine with changing as long it is documented, but I don`t see any documentation in the PR.
The other question would be, why we should keep the OBS_API_SETUP variable at all. From my POV we also could remove it from /etc/sysconfig/obs-server completely as it (IMHO) makes no sense to have two mechanisms to disable obsapisetup but two preconditions to enable it.
I guess this is in there for historical reasons and I would like to hear @adrianschroeter 's opinion.

Reply:
After a personal discussion with @adrianschroeter we agreed on the following:

* Removing OBS_API_SETUP from sysconfig is a good idea
* ATM we will keep 'OBS_API_SETUP' because we would need a migration scenario for that which would add additional complexity to the current change to systemd
* After an update, the state (enabled/disabled) should not be changed
  * (IMHO) We have to ensure this with integration tests